services/frontend-service/src/ui/TopAppBar/TopAppBar.tsx

```diff
@@ -16,14 +16,23 @@ along with kuberpult.  If not, see <http://www.gnu.org/licenses/>.
 Copyright 2021 freiheit.com*/
 import { MDCTopAppBar } from '@material/top-app-bar';
 
-import { Button } from '../components/button';
 import { Textfield } from '../components/textfield';
-import { useEffect, useRef } from 'react';
+import { useCallback, useEffect, useRef } from 'react';
+import { SideBar } from '../SideBar/SideBar';
+import { Button } from '../components/button';
 
 export const TopAppBar: React.FC = () => {
     const control = useRef<HTMLDivElement>(null);
     const MDComponent = useRef<MDCTopAppBar>();
 
+    const displaySideBar = useCallback(() => {
+        const sideBarRef = document.getElementsByClassName('mdc-drawer--hidden')[0];
+        if (sideBarRef?.classList.contains('mdc-drawer--hidden')) {
```

Comment:
This is not react way. I think this would be better:

```
const [showSidebar, setShowSidebar] = useState(false);
const displaySideBar = useCallback(() => setShowSidebar(prev => !prev), [setShowSidebar]);
```

and then later have either this:

```
{showSidebar && <SideBar />}
```

or like this:
```
<SideBar visible={showSidebar} />}
```

Reply:
I used that way at first and I came across some errors stating that too many re-renders were being done. This was a topic I actually wanted to discuss

Reply:
I fixed this and now it uses solely useState and reference passing